rules.py

```diff
@@ -0,0 +1,12 @@
+from num_gen import substitute
+from collections import Counter
+
+
+def rules(list):
```

Comment:
Same as above, do not use list as variable/argument name since it is a keyword.
Try to use some more meaningful name.